test/unit/variable-scope.test.js

```diff
@@ -631,6 +631,194 @@ describe('VariableScope', function () {
+            it('should resolve all variables in string', function () {
+                var scope = new VariableScope({
+                    values: [{
+                        key: 'name',
+                        value: 'Cooper'
+                    }, {
+                        key: 'job',
+                        value: 'Postman'
+                    }]
+                });
+
+                expect(scope.replaceIn('I am {{name}} and I work at {{job}}'))
+                    .to.equal('I am Cooper and I work at Postman');
+            });
+
+            it('should resolve all variables in arrays', function () {
+                var scope = new VariableScope({
+                    values: [{
+                        key: 'name',
+                        value: 'Cooper'
+                    }, {
+                        key: 'job',
+                        value: 'Postman'
+                    }]
+                });
+
+                expect(scope.replaceIn([])).to.eql([]);
+                expect(scope.replaceIn(['text'])).to.eql(['text']);
+                expect(scope.replaceIn(['{{name}}'])).to.eql(['Cooper']);
+                expect(scope.replaceIn([{ name: '{{name}}' }, { job: '{{job}}' }]))
+                    .to.eql([{ name: 'Cooper' }, { job: 'Postman' }]);
+            });
+
+            it('should resolve variables in a user defined type instance', function () {
+                var scope = new VariableScope({
+                        values: [{
+                            key: 'var',
+                            value: 'value'
+                        }]
+                    }),
+                    MyVar = function (val) {
+                        this.val = val;
+                    },
+                    myVar = new MyVar('{{var}}');
+
+                expect(scope.replaceIn(myVar)).to.eql({ val: 'value' });
+            });
+
+            it('should not resolve disabled variables', function () {
+                var scope = new VariableScope({
+                    values: [{
+                        key: 'name',
+                        value: 'Cooper'
+                    }, {
+                        key: 'job',
+                        value: 'Postman',
+                        disabled: true
+                    }]
+                });
+
+                expect(scope.replaceIn('I am {{name}} and I work at {{job}}'))
+                    .to.equal('I am Cooper and I work at {{job}}');
+            });
+
+            it('should not resolve variable templates with spaces', function () {
+                var scope = new VariableScope({
+                    values: [{
+                        key: 'place',
+                        value: 'Bangalore'
+                    }]
+                });
+
+                expect(scope.replaceIn('{{place}} is not {{ place }}'))
+                    .to.equal('Bangalore is not {{ place }}');
+            });
+
+            it('should be able to resolve variables deeply', function () {
```

Comment:
We don't use programmatic sets like this, better test such conditions individually.
We can remove this test.